### ISSUE TITLE
Can not find the Point-wise Mesh Euclidean Distance measurement code

### ISSUE THREAD

**Issue by mikecheninoulu** [jiashunwang/Neural-Pose-Transfer#9]
Hi, thanks for your good work, but we cannot find the protocols to measure the performance, especially the code of Point-wise Mesh Euclidean Distance for supervised and unsupervised (only list)

Could you provide one?

**Comment by jiashunwang** [comment]
Hi,
we align the meshes first by letting all the points minus their own bbox centers and then use MSE loss to calculate the error of GT and result.

**Comment by jiashunwang** [comment]
Hope this code helps you.
![Uploading evaluate.png…]()

**Comment by jiashunwang** [comment]
![evaluate](https://user-images.githubusercontent.com/51504170/96441527-3a46fa80-123c-11eb-8b8b-b8ae49f57d66.png)
Hope this helps you.

**Comment by mikecheninoulu** [comment]
Got it! Thank you so much for your help!

**Comment by mikecheninoulu** [comment]
Hi, however, when I implemented your code to reproduce the results, the results I got are away worse than yours. The supervised and unsupervised results are similar, (0.2608 and 0.2526 with 72 subjects for each), could you figure out what is the problem with my code?

Here is the model I used:
from model import NPT
net_G=NPT()
net_G.cuda()
net_G.load_state_dict(torch.load('original_169.model'))

Here is the implemented code:
![Screenshot from 2020-10-20 10-39-32](https://user-images.githubusercontent.com/17347668/96562201-c995f180-12c0-11eb-834e-564b424e8c8c.png)

I tried different models, max_pooling and original model and I also trained them myself, but the results are more or less the same (around 0.25 which is far away from yours 0.00011), thank for you help!

**Comment by jiashunwang** [comment]
Hi,
Thank you for your question.
I think there is a problem you missed. If you shuffle the identiy mesh, then the gt should also be shuffled in the same way, cause the output vertices is at the same order as the identity mesh. The output may looks like GT visually but has huge error in your code.

发件人: mikecheninoulu<mailto:notifications@github.com>
发送时间: 2020年10月20日 16:43
收件人: jiashunwang/Neural-Pose-Transfer<mailto:Neural-Pose-Transfer@noreply.github.com>
抄送: Jiashun Wang<mailto:jiashun_wang@outlook.com>; Comment<mailto:comment@noreply.github.com>
主题: Re: [jiashunwang/Neural-Pose-Transfer] Can not find the Point-wise Mesh Euclidean Distance measurement code (#9)


Hi, however, when I implemented your code to reproduce the results, the results I got are away worse than yours. The supervised and unsupervised results are similar, (0.2608 and 0.2526 with 72 subjects for each), could you figure out what is the problem with my code?

Here is the model I used:
from model import NPT
net_G=NPT()
net_G.cuda()
net_G.load_state_dict(torch.load('original_169.model'))

Here is the implemented code:
[Screenshot from 2020-10-20 10-39-32]<https://user-images.githubusercontent.com/17347668/96562201-c995f180-12c0-11eb-834e-564b424e8c8c.png>

I tried different models, max_pooling and original model and I also trained them myself, but the results are more or less the same (around 0.25 which is far away from yours 0.00011), thank for you help!

―

**Comment by mikecheninoulu** [comment]
> Hi, Thank you for your question. I think there is a problem you missed. If you shuffle the identiy mesh, then the gt should also be shuffled in the same way, cause the output vertices is at the same order as the identity mesh. The output may looks like GT visually but has huge error in your code.

Thank you so much for your fast reply!  Exactly as you said, now I can make it close to your results.
WIth your pretrained max_pooling model, the supervised and unsupervised results are 0.000723 and 0.001367.
WIth your pretrained original model, the supervised and unsupervised results are 0.03597 and 0.03577079897669046.

It seems like there is still a problem of using original model. I only changed these:
from 
```
from model_max_pool import NPT
net_G=NPT()
net_G.cuda()
net_G.load_state_dict(torch.load('maxpool.model'))
```

to 
```
from model import NPT
net_G=NPT()
net_G.cuda()
net_G.load_state_dict(torch.load('original.model'))
```

**Comment by jiashunwang** [comment]
Hi
I think the result we provide is in the situation which identity and pose meshes are in different orders. If they are the at same, I think for original_model, it may fail. So if you want to test original_model, I recommend you shuffle the mesh order first and then check the number.

**Comment by mikecheninoulu** [comment]
> Hi
> I think the result we provide is in the situation which identity and pose meshes are in different orders. If they are at the same, I think for original_model, it may fail. So if you want to test original_model, I recommend you shuffle the mesh order first and then check the number. It is convenient if you just change the pose. But it also works if you shuffle all of them, but be careful for the identity and GT should be same.

Thank you so much for your help, problem solved! It's really a pleasure to discourse with you. Do you have WeChat, may you add me as friend? Mine: cc930315 :)
